config/params/config_utils_prod.go

```diff
@@ -7,26 +7,24 @@ import (
 	"github.com/mohae/deepcopy"
 )
 
-var beaconConfig = MainnetConfig()
-
 // BeaconConfig retrieves beacon chain config.
 func BeaconConfig() *BeaconChainConfig {
-	return beaconConfig
+	return Registry.GetActive()
 }
 
 // OverrideBeaconConfig by replacing the config. The preferred pattern is to
 // call BeaconConfig(), change the specific parameters, and then call
 // OverrideBeaconConfig(c). Any subsequent calls to params.BeaconConfig() will
 // return this new configuration.
 func OverrideBeaconConfig(c *BeaconChainConfig) {
-	beaconConfig = c
+	Registry.active = c
```

Comment:
Very important note here - `OverrideBeaconConfig` should *only* be used by tests. It bypasses SetActive (which calls Register, which calls Add), so configs set to active in this way will not update the registry, which means that VersionedUnmarshaler won't be able to detect their configuration. I wanted to delete `OverrideBeaconConfig()` and `BeaconChainConfig()` entirely and replace it with Registry methods, but made this compromise to break up the change across multiple PRs.

Reply:
That is not the case though, this is used in `configureTestnet`. Is that intentional ?

Reply:
please check out updated diff